sklearn/ensemble/_stacking.py

```diff
@@ -173,37 +180,50 @@ def fit(self, X, y, sample_weight=None):
             else:
                 self.named_estimators_[name_est] = "drop"
 
-        # To train the meta-classifier using the most data as possible, we use
-        # a cross-validation to obtain the output of the stacked estimators.
-
-        # To ensure that the data provided to each estimator are the same, we
-        # need to set the random state of the cv if there is one and we need to
-        # take a copy.
-        cv = check_cv(self.cv, y=y, classifier=is_classifier(self))
-        if hasattr(cv, "random_state") and cv.random_state is None:
-            cv.random_state = np.random.RandomState()
-
         self.stack_method_ = [
             self._method_name(name, est, meth)
             for name, est, meth in zip(names, all_estimators, stack_method)
         ]
-        fit_params = (
-            {"sample_weight": sample_weight} if sample_weight is not None else None
-        )
-        predictions = Parallel(n_jobs=self.n_jobs)(
-            delayed(cross_val_predict)(
-                clone(est),
-                X,
-                y,
-                cv=deepcopy(cv),
-                method=meth,
-                n_jobs=self.n_jobs,
-                fit_params=fit_params,
-                verbose=self.verbose,
+
+        # To train the meta-classifier using the most data as possible, we use
+        # a cross-validation to obtain the output of the stacked estimators.
```

Comment:
Since this comment only applies to the `cv != "prefit"` case, may we move this after the `else` statement?

(There is no actual cross validation when `cv == "prefit")